### PR TITLE
Microvm helpers for interactive usage

### PR DESCRIPTION
**Pull request by pb8o**
## Changes

Some unrelated changes included:
- a fix for the devctr sanity build
- remove some dead code

## Reason

This enables interactive use of Microvms, will follow up with a sandbox helper script.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by pb8o** [comment]
Reopening as it was closed by mistake

**Review by sudanl0 — DISMISSED** [review]

**Review by bchalios — COMMENTED** [review]
That looks awesome! Just one comment. Would it be easy to add some documentation on how to use it to launch an interactive session and small examples with what you can do with it?

**Comment by pb8o** [comment]
> That looks awesome! Just one comment. Would it be easy to add some documentation on how to use it to launch an interactive session and small examples with what you can do with it?

Actually I had some docs written for a different change, but they belong here. I will update the PR.

**Review by bchalios — DISMISSED** [review]
LGTM! 🍻

**Review by sudanl0 — DISMISSED** [review]

File: tests/README.md [hunk 1]
```diff
@@ -192,32 +192,10 @@ source tree. This directory is bind-mounted in the container and used as a
 local image cache.
 
 `Q5:`
-*Is there a way to speed up integration tests execution time?*
```
Comment (sudanl0) [inline tests/README.md:195]: Was the FAQ removed because it was inaccurate?

Comment (pb8o) [reply tests/README.md:195]: Fixed

**Review by bchalios — APPROVED** [review]

File: tests/host_tools/memory.py [hunk 2]
```diff
@@ -54,7 +54,10 @@ def run(self):
         """
 
         guest_mem_bytes = self._vm.mem_size_bytes
-        ps = psutil.Process(self._vm.jailer_clone_pid)
+        try:
+            ps = psutil.Process(self._vm.jailer_clone_pid)
+        except psutil.NoSuchProcess:
```
Comment (sudanl0) [inline tests/host_tools/memory.py:59]: note for future reference: if memory monitor runs after the test was killed, the process probably ran for a very low amount of time in which case collecting the overhead doesn't really make sense.

**Review by sudanl0 — APPROVED** [review]

**Pull request merged**